### PR TITLE
fix: When token cache is invalid, don't raise `TypeError`

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/credentials.py [hunk 1]
```diff
@@ -288,7 +288,10 @@ def scopes(self) -> list[str]:
     def _refresh_access_token(self) -> tuple[str, float]:
         # First check if there is a serialized token cached on disk.
         if accounts := self.__app.get_accounts():
-            credentials = self.__app.acquire_token_silent(scopes=self.__scopes, account=accounts[0])
+            credentials = self.__app.acquire_token_silent_with_error(scopes=self.__scopes, account=accounts[0])
+            if "error" in credentials:
```
Comment (erlendvollset) [inline cognite/client/credentials.py:292]: Perhaps a retry loop with exponential backoff? There is some retryutil added recently

Comment (erlendvollset) [reply cognite/client/credentials.py:292]: Or maybe we should make this exception something the existing retry logic can.

Comment (erlendvollset) [reply cognite/client/credentials.py:292]: And I would also suggest being more specific about which error to retry. Is there more info in the error object?

Comment (erlendvollset) [reply cognite/client/credentials.py:292]: I see now this is only when using the persisted token, so I guess a single retry should suffice and that you want to do it regardless of the error mode. LGTM!

Comment (erlendvollset) [reply cognite/client/credentials.py:292]: But we should use the same for logic for the other login flows

Comment (doctrino) [reply cognite/client/credentials.py:296]: Done

File: cognite/client/credentials.py [hunk 2]
```diff
@@ -209,9 +209,13 @@ def scopes(self) -> list[str]:
...
         if accounts := self.__app.get_accounts():
-            credentials = self.__app.acquire_token_silent(scopes=self.__scopes, account=accounts[0])
-        # If not, we acquire a new token using device code auth flow:
+            credentials = self.__app.acquire_token_silent_with_error(scopes=self.__scopes, account=accounts[0])
         else:
+            credentials = {}
+
+        # If not, we acquire a new token using device code auth flow:
+        # or the token might be expired, so we try to refresh it.
+        if not accounts or "error" in credentials:
```
Comment (haakonvt) [inline cognite/client/credentials.py]: MSAL docs says this can still return `None` when:
-> None when there is simply no token in the cache.

**Review by haakonvt — APPROVED** [review]
Pushed a few suggested commits after 🆗 -ing on slack.

**Pull request merged**